packages/middleware-endpoint/src/adaptors/getEndpointFromInstructions.ts

```diff
@@ -39,9 +39,19 @@ export const getEndpointFromInstructions = async <
   context?: HandlerExecutionContext
 ): Promise<EndpointV2> => {
   if (!clientConfig.endpoint) {
-    const endpointFromConfig = await getEndpointFromConfig(clientConfig.serviceId || "");
```

Comment:
the issue was that this call involved a call to `loadConfig` for every request, which _calls_ `memoize` rather than calling a memoized function, leading to repeated calls to fs.

Reply:
this delayed every request by around 50ms in my testing

Reply:
edit: "slurpFile" has a cache, but it is several functions/modules deep in terms of unnecessary calls